### PR TITLE
create a new route for slot status

### PR DESCRIPTION
**Pull request by galenguyer**
why do we return an array of strings, that's just awful

now we can return the same data but in json

### CODE REVIEW

File: src/routes.rs [hunk 1]
```diff
@@ -66,3 +71,16 @@ async fn health(data: web::Data<Mutex<AppData>>) -> impl Responder {
         temp: temperature,
     })
 }
+
+#[get("/slots")]
+async fn get_slots(data: web::Data<Mutex<AppData>>) -> impl Responder {
+    let slots = machine::get_slots(data.lock().unwrap().config.clone());
+    let temp = machine::get_temperature(data.lock().unwrap().config.clone());
+
+    let temp = temp * (9.0/5.0) + 32.0;
```
Comment (Mstrodl) [inline src/routes.rs]: If we're redesigning, can we just report celsius?

File: src/routes/machine.rs [hunk 2]
```diff
@@ -45,6 +46,22 @@ pub fn get_slots(config: ConfigData) -> Vec<String> {
...
 }
 
+#[derive(Serialize)]
+pub struct SlotStatus {
+    pub id: String,
+    pub number: i32,
+    pub stocked: bool,
+}
+pub fn get_slots(config: ConfigData) -> Vec<SlotStatus> {
+    (1..=config.slot_ids.len()).zip(config.slot_ids).map(|(n, i)| {
```
Comment (Mstrodl) [inline src/routes/machine.rs]: Can't we `config.slot_ids.iter().map`? Or is there no way to map with index?
Either way, this feels very messy and we should at least have more than single-letter variable names

**Review by Mstrodl — CHANGES_REQUESTED** [review]

File: src/routes.rs (hunk 1 above)
Comment (galenguyer) [reply src/routes.rs]: Sure

File: src/routes/machine.rs (hunk 2 above)
Comment (galenguyer) [reply src/routes/machine.rs]: `.iter().enumerate()` is what we wanted. Didn't realize that existed, oops

**Review by Mstrodl — APPROVED** [review]
:shipit:

**Pull request merged**